lib/encoder.js

```diff
@@ -1442,7 +1442,7 @@ function handleBufferRef(buffer) {
  * @private
  */
 function decodeCollectionLengthV3(bytes, offset) {
-  return bytes.readUInt32BE(offset);
+  return bytes.readInt32BE(offset);
```

Comment:
Would you like to adapt `getLengthBufferV3()` just for the sake of symmetry, as part of this patch?